File: src/main/java/destiny/core/chinese/ziwei/HouseHourBranchImpl.java
/**
 * Created by smallufo on 2017-04-13.
 */
package destiny.core.chinese.ziwei;

import destiny.core.Gender;
import destiny.core.calendar.SolarTerms;
import destiny.core.chinese.Branch;
import destiny.core.chinese.StemBranch;

import java.util.Optional;

/**
 * 時支 -> 地支
 */
public abstract class HouseHourBranchImpl extends HouseAbstractImpl<Branch> {

  HouseHourBranchImpl(ZStar star) {
    super(star);
  }

  @Override
  public Branch getBranch(StemBranch lunarYear, StemBranch solarYear, Branch monthBranch, int finalMonthNumForMonthStars, SolarTerms solarTerms, int days, Branch hour, int set, Gender gender, boolean leap, int prevMonthDays, Optional<Branch> predefinedMainHouse, ZContext context) {
    return getBranch(hour);
  }
}

File: src/main/java/destiny/core/chinese/ziwei/HouseHourBranchImpl.kt
/**
 * Created by smallufo on 2017-04-13.
 */
package destiny.core.chinese.ziwei

import destiny.core.chinese.Branch

/**
 * 時支 -> 地支
 */
abstract class HouseHourBranchImpl internal constructor(star: ZStar) : HouseAbstractImpl<Branch>(star) {

  override fun getBranch(context: HouseCalContext): Branch {
    return getBranch(context.hour)
  }
}
